bees/mumblebee/mumblebee.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	return outs
+}
+
+func (mod *MumbleBee) move(channel string) {
+	channel = strings.TrimSpace(channel)
+	if len(channel) > 0 {
+		channels := strings.Split(channel, "/")
+		result := mod.client.Channels.Find(channels...)
+		if result != nil {
+			mod.client.Self.Move(result)
+			mod.Logln("Moved to channel:", channel)
+		} else {
+			mod.Logln("Moved to channel:", channel, " failed. (channel not found)")
+		}
+	}
+}
+
+// Run executes the Bee's event loop.
+func (mod *MumbleBee) Run(eventChan chan bees.Event) {
+	if len(mod.server) == 0 {
+		return
+	}
+
+	// channel signaling mumble connection status
+	mod.connectedState = make(chan bool)
+
+	host, port, err := net.SplitHostPort(mod.server)
+	if err != nil {
+		host = mod.server
+		port = strconv.Itoa(gumble.DefaultPort)
+	}
+
+	config := gumble.NewConfig()
+	config.Username = mod.user
+	config.Password = mod.password
+	address := net.JoinHostPort(host, port)
+
+	var tlsConfig tls.Config
+
+	if mod.insecure {
+		tlsConfig.InsecureSkipVerify = true
+	}
+
+	config.Attach(gumbleutil.Listener{
+		TextMessage: func(e *gumble.TextMessageEvent) {
+			ev := bees.Event{
+				Bee:  mod.Name(),
+				Name: "message",
+				Options: []bees.Placeholder{
+					{
+						Name:  "channels",
+						Type:  "[]string",
+						Value: e.Channels,
+					},
+					{
+						Name:  "user",
+						Type:  "string",
+						Value: e.Sender.Name,
+					},
+					{
+						Name:  "text",
+						Type:  "string",
+						Value: e.Message,
+					},
+				},
+			}
+			eventChan <- ev
+		},
+	})
+	config.Attach(gumbleutil.Listener{
+		Disconnect: func(e *gumble.DisconnectEvent) {
+			mod.Logln("DisconnectEvent")
+			go func() {
+				mod.connectedState <- false
+			}()
+		},
+	})
+	config.Attach(gumbleutil.Listener{
+		Connect: func(e *gumble.ConnectEvent) {
+			mod.Logln("ConnectEvent")
+			go func() {
+				mod.connectedState <- true
+			}()
+		},
+	})
+
+	/*/ ToDo: Use own client certs
+	if *certificateFile != "" {
+		if *keyFile == "" {
+			keyFile = certificateFile
+		}
+		if certificate, err := tls.LoadX509KeyPair(*certificateFile, *keyFile); err != nil {
+			fmt.Printf("%s: %s\n", os.Args[0], err)
+			os.Exit(1)
+		} else {
+			tlsConfig.Certificates = append(tlsConfig.Certificates, certificate)
+		}
+	}
+	//*/
+
+	config.Attach(gumbleutil.AutoBitrate)
+
+	connecting := false
+	disconnected := true
+	waitForDisconnect := false
+	for {
+		// loop on IRC connection events
+		if disconnected {
+			if waitForDisconnect {
+				return
+			}
+
+			if !connecting {
+				connecting = true
+				mod.Logln("Connecting to Mumble:", mod.server)
+				mod.client, err = gumble.DialWithDialer(new(net.Dialer), address, config, &tlsConfig)
+
+				mod.Logln("Connected to:", mod.server)
+				if err != nil {
+					mod.Logln("Failed to connect to Mumble:", mod.server, err)
+					connecting = false
+				}
+			}
+		}
+		select {
+		case status := <-mod.connectedState:
+			mod.Logln("status case")
```

Comment:
Probably best to remove this and the other ```status xxx``` debug/development log-messages below so we don't spam our user's logs.